ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/transaction/Rejections.scala

```diff
@@ -24,22 +24,24 @@ private[transaction] class Rejections(metrics: Metrics) {
 
   private final val logger = ContextualizedLogger.get(getClass)
 
-  def buildRejectionStep[A](
+  def buildImmediateRejectionStep[A](
```

Comment:
I'm not sure what "immediate" means here. Could you please explain?

Reply:
It means that the step has to be the result of the validation. Perhaps `buildConclusiveRejection`? Or even `reject`, so that it's clear that it has side effects (metrics and logging).

Reply:
Yes, I think that if it records something, even metrics, we need to capture that in the name so it's clear it's not pure. `reject` works well IMO.

Are there any functions that build a rejection that could be thrown away later?

Reply:
Not so far but see [here](https://github.com/digital-asset/daml/pull/10454#pullrequestreview-720118302).